### PR TITLE
Candidate can see rejection reasons from Apply1 application

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

As a... candidate
I need to... see why my previous application was rejected
So that... I can make amendments when I Apply again

## Changes proposed in this pull request

- if an application form is apply2 and their appy1 has rejections. Show the reasons

image

## Guidance to review

Should some of the controller logic be pulled down to the model? i.e. 
`@apply_1_application_form = current_candidate.application_forms.apply_1.last`

## Link to Trello card

https://trello.com/c/VQV2ZylK/1392-candidates-can-see-the-reasons-why-their-previous-application-was-rejected-when-they-apply-again

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/controllers/candidate_interface/application_form_controller.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ class ApplicationFormController < CandidateInterfaceController
     def show
       @application_form_presenter = CandidateInterface::ApplicationFormPresenter.new(current_application)
       @application_form = current_application
+      @previous_application_form = current_candidate.application_forms.where.not(id: current_application.id).order(id: :desc).last
```
Comment (tvararu) [inline app/controllers/candidate_interface/application_form_controller.rb:9]: Steve's working on an association between the two applications, probably `.original_application_form`. We can still merge this, and update it later, as long as we make a note somewhere not to forget 👍

File: app/services/feature_flag.rb [hunk 2]
```diff
@@ -8,6 +8,7 @@ class FeatureFlag
   ].freeze
 
   TEMPORARY_FEATURE_FLAGS = %w[
+    apply_again_rejection_reasons
```
Comment (tvararu) [inline app/services/feature_flag.rb:11]: Let's lump this into `apply_again` in general, we shouldn't need another flag

```suggestion
```

File: app/views/candidate_interface/application_form/show.html.erb [hunk 3]
```diff
@@ -17,6 +17,10 @@
   <div class="govuk-grid-column-two-thirds">
     <p class="govuk-hint govuk-!-margin-bottom-8 govuk-!-margin-top-2"><%= @application_form_presenter.updated_at %></p>
 
+    <% if FeatureFlag.active?('apply_again_rejection_reasons') && @application_form.apply_2? && @previous_application_form.application_choices.rejected.present?  %>
```
Comment (tvararu) [inline app/views/candidate_interface/application_form/show.html.erb:20]:
```suggestion
    <% if FeatureFlag.active?('apply_again') && @application_form.apply_2? && @previous_application_form.application_choices.rejected.present?  %>
```

Comment (tvararu) [reply app/views/candidate_interface/application_form/show.html.erb:20]: The `.apply_2?` could also become `.original_application_form.exists?` or such, later

File: spec/system/candidate_interface/candidate_can_see_their_rejection_reasons_on_apply_again_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    and_i_am_signed_in
+    and_i_have_an_apply1_application_with_3_rejections
+    and_i_have_started_my_apply_again_application_form
+
+    when_i_visit_my_apply_again_application_form
+    then_i_can_see_my_previous_rejection_reasons
+  end
+
+  def given_apply_again_rejection_reasons_is_active
+    FeatureFlag.activate('apply_again_rejection_reasons')
```
Comment (tvararu) [inline spec/system/candidate_interface/candidate_can_see_their_rejection_reasons_on_apply_again_spec.rb:15]:
```suggestion
    FeatureFlag.activate('apply_again')
```

File: spec/system/candidate_interface/candidate_can_see_their_rejection_reasons_on_apply_again_spec.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    given_apply_again_rejection_reasons_is_active
+    and_i_am_signed_in
+    and_i_have_an_apply1_application_with_3_rejections
+    and_i_have_started_my_apply_again_application_form
+
+    when_i_visit_my_apply_again_application_form
+    then_i_can_see_my_previous_rejection_reasons
+  end
+
+  def given_apply_again_rejection_reasons_is_active
```
Comment (tvararu) [inline spec/system/candidate_interface/candidate_can_see_their_rejection_reasons_on_apply_again_spec.rb:14]:
```suggestion
  def given_apply_again_is_active
```

File: spec/system/candidate_interface/candidate_can_see_their_rejection_reasons_on_apply_again_spec.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe 'Candidate can see their rejection reasons on apply again' do
+  scenario 'when a candidate visits their apply again application form they can see apply1 rejection reasons' do
+    given_apply_again_rejection_reasons_is_active
```
Comment (tvararu) [inline spec/system/candidate_interface/candidate_can_see_their_rejection_reasons_on_apply_again_spec.rb:5]:
```suggestion
    given_apply_again_is_active
```

**Comment by fofr** [comment]
Screen Shot 2020-05-01 at 13 49 19

Can we remove the rogue whitespace between bottom of card and the left border?

File: app/views/candidate_interface/application_form/show.html.erb (hunk 3 above)
Comment (fofr) [inline app/views/candidate_interface/application_form/show.html.erb:20]: Could this logic live within the component itself? (Should it @tvararu?)

**Pull request merged**